torch/CMakeLists.txt

```diff
@@ -182,6 +181,29 @@ set(TORCH_SRCS
   ${TORCH_SRC_DIR}/csrc/utils/variadic.cpp
   )
 
+if (NOT WIN32)
+  add_definitions(-DUSE_CPU_FUSER)
```

Comment:
Yeah, this isn't going to fly. If you were only using the macros from C++ files, it'd be OK, but you've got the macros in headers, which means that you need to make sure that they get durably set, even if other projects include your header in question.

There are two ways to do this:

1. Use `target_compile_definitions`. Example in `caffe2/CMakeLists.txt`

```
  # NB: This must be target_compile_definitions, not target_compile_options,
  # as the latter is not respected by nvcc
  target_compile_definitions(caffe2 PUBLIC "AT_CORE_STATIC_WINDOWS=1")
```

This will be propagated to downstream projects IFF they use cmake to do the build, and use the installed cmake config file. This is somewhat error prone.

2. Write out a macro file, config.h style, which specifies if we built with or without the support. Example `aten/src/ATen/Config.h.in`. A quick grep suggests we don't have anything equivalent in torch at the moment.

Reply:
Thanks Ed, this is really helpful.  

Since you're the expert, would you tell me which approach you'd prefer? I'm up for either and it sounds like you're leaning towards (1)?